models/sudoswap/ethereum/sudoswap_ethereum_pools.sql

```diff
@@ -0,0 +1,275 @@
+{{
+ config(
+       alias='pools',
+       partition_by = ['block_date'],
+       materialized = 'incremental',
+       file_format = 'delta',
+       incremental_strategy = 'merge',
+       unique_key = ['pool_address', 'nft_contract_address', 'creator_address'],
+       post_hook='{{ expose_spells(\'["ethereum"]\',
+                                   "project",
+                                   "sudoswap",
+                                   \'["niftytable"]\') }}'
+      )
+}}
+
+WITH
+  pairs_created AS (
+    SELECT
+      CASE
+        WHEN _bondingCurve = '0x5b6ac51d9b1cede0068a1b26533cace807f883ee' THEN 'linear_bonding'
+        ELSE 'exp_bonding'
+      END as pricing_type,
+      _delta / 1e18 as delta,
+      CASE
+        WHEN _poolType = 0 THEN 'token'
+        WHEN _poolType = 1 THEN 'nft'
+        WHEN _poolType = 2 THEN 'trade'
+      END AS pool_type,
+      _spotPrice / 1e18 AS spot_price,
+      _nft AS nft_contract_address,
+      _initialNFTIDs AS nft_ids,
+      _fee AS initialfee,
+      output_pair AS pair_address,
+      call_block_time AS block_time,
+      tx.FROM AS creator_address,
+      date_trunc('day', now()) - date_trunc('day', call_block_time) AS days_passed,
+      cardinality(_initialNFTIDs) AS initial_nft_count,
+      tx.value / 1e18 AS initial_eth
+    FROM
+      {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }} cre
+      INNER JOIN {{ source('ethereum','transactions') }} tx ON tx.hash = cre.call_tx_hash
+    WHERE
+      call_success
+    {% if is_incremental() %}
```

Comment:
approach for incremental logic:
1. use in where clause on the table in the 'from' -- i.e. ```cre``` alias in this query
2. use in the join conditions on joined tables -- i.e. ```trx``` alias in this query (use in 'and' logic rather than where)

Reply:
cool, in that case its not needed here right? will remove